### DISCUSSION TITLE
Error on upgrade from Webpack 5.85.1 to 5.86.0

### DISCUSSION THREAD

**Discussion by eamodio** [webpack/webpack#17354] [Q&A] [answered]
# Bug report

**What is the current behavior?**

When upgrading from Webpack `5.85.1` to `5.86.0` my build started throwing the following errors.

```
ERROR in   Error: Cannot use import statement outside a module
    
    - node:vm:100 new Script
      node:vm:100:7
    
    - index.js:138 HtmlWebpackPlugin.evaluateCompilationResult
      [vscode-gitlens]/[html-webpack-plugin]/index.js:138:22
    
    - index.js:324 
      [vscode-gitlens]/[html-webpack-plugin]/index.js:324:26
    
    - async Promise.all
    
    - async Promise.all
```

My Webpack config can be seen here: https://github.com/gitkraken/vscode-gitlens/blob/dffefeb4116554d7d36e6f1537d9090266c19e01/webpack.config.js

**What is the expected behavior?**

To not error like `5.85.1`

**Other relevant information:**
webpack version:  `5.86.0`
Node.js version: `v18.16.0`
Operating System: Windows 11 Pro 22H2 `22631.1830`
Additional tools:

**Comment by alexander-akait** [comment]
I think somewhere was used `__non_webpack_require__` and you have `type: "module"` or `"javascript/esm"`, so now it is `import { createRequire }` from "module"`, previously we generate `require`, but it was a bug, if you provide reprducible test repo I will show you where and how to fix it

**Reply by alexander-akait** [reply]
Hm, can I use `vscode-gitlens` to reproduce the problems?

**Reply by alexander-akait** [reply]
Reproduced https://github.com/jantimon/html-webpack-plugin/actions/runs/5214504235/jobs/9410850818?pr=1795, looking for a fix

**Reply by eamodio** [reply]
Sorry for the delay in getting back to you, yes, it is reproducible with `vscode-gitlens` if you bump the `webpack` version. But it looks like you got a reproduction -- thank you!

**Reply by alexander-akait** [reply]
yeah, don't know why html webpack plugin has
```
var _ = __non_webpack_require__(' + JSON.stringify(require.resolve('lodash')) + ');
```

we really need rewrite it for better compatibility with webpack

**Reply by alexander-akait** [reply]
@eamodio Sorry for delay, yesterday was a problem on npm side, so I can't publish https://github.com/jantimon/html-webpack-plugin, today there there are no problems, so please try https://github.com/jantimon/html-webpack-plugin/blob/main/CHANGELOG.md#552-2023-06-08

**Reply by eamodio** [reply, marked as answer]
@alexander-akait Thank you! Yup, that fixed it!
